### PR TITLE
cleanup: move cli script to separate file

### PR DESCRIPTION
**Pull request by philipyoo**
Moving command line script to another file in anticipation for building a UI portion.
Added a way for a user to define location of an AWS credentials file. User must use `AWS_SHARED_CREDENTIALS_FILE`, or we can change this to something shorter as well

### CODE REVIEW

**Comment by philipyoo** [comment]
@bennettbuchanan awesome thank you! Added edits :)

**Comment by rahulreddy** [comment]
This is good. As Bennett mentioned, please make sure `AWS_SHARED_CREDENTIALS_FILE` is optional but it gets first priority if it is defined. We will fallback to the `~/.aws/credentials` if the env var is not defined.

File: bin/replication.js [hunk 1]
```diff
@@ -61,7 +54,10 @@ function _buildResourcePolicy(source, target) {
 }
 
 function _setupS3Client(transport, endpoint, profile) {
-    const credentials = new SharedIniFileCredentials({ profile });
+    const filename = process.env.AWS_SHARED_CREDENTIALS_FILE ||
+        `${os.homedir()}/.aws/credentials`;
+    const credentials = new SharedIniFileCredentials({ profile, filename });
```
Comment (jonathan-gramain) [inline bin/replication.js]: I think you can make `credentials` a global variable, that would avoid reconstructing it for different clients. Globals are usually fine to expose global configuration values.

File: bin/replication.js [hunk 2]
```diff
@@ -115,8 +114,10 @@ class _SetupReplication {
             'default' : sourceProfile;
         const verifyTargetProfile = targetProfile === undefined ?
             'default' : targetProfile;
-        source.auth.vault.host = 'localhost';
-        source.s3.host = 'localhost';
+
+        // TODO: Federation fix made #1125?
+        // source.auth.vault.host = 'localhost';
+        // source.s3.host = 'localhost';
         const destHost = this.destHosts.pickHost().host;
```
Comment (jonathan-gramain) [inline bin/replication.js]: If I'm not mistaken, you should pick both `host` and `port` from the result of `pickHost`, then construct the endpoint string as `host:port`.

Comment (philipyoo) [inline bin/replication.js]: For this, Rahul wanted to use s3-nginx port 80 https://github.com/scality/backbeat/pull/62#discussion_r133603964

File: bin/replication.js (hunk 1 above)
Comment (philipyoo) [inline bin/replication.js]: Thanks! Ya that sounds much better :) Will make this change and rebase soon

File: bin/replication.js (hunk 2 above)
Comment (rahulreddy) [inline bin/replication.js]: Yes, for now we are hardcoding the destination S3/IAM ports.

Comment (jonathan-gramain) [inline bin/replication.js]: OK, I understand, I think it deserves a comment explaining the reason though, and/or a FIXME for later.

**Pull request closed without merging**